services/elasticache/pkg/resource/cache_parameter_group/custom_api.go

```diff
@@ -48,13 +91,57 @@ func (rm *resourceManager) describeCacheParameters(
 			break
 		}
 		for _, p := range response.Parameters {
-			name := *p.ParameterName
-			value := *p.ParameterValue
-			sp := svcapitypes.ParameterNameValue{
-				ParameterName:  &name,
-				ParameterValue: &value,
+			var name *string = nil
+			if p.ParameterName != nil {
+				v := *p.ParameterName
+				name = &v
+			}
+			var value *string = nil
+			if p.ParameterValue != nil {
+				v := *p.ParameterValue
+				value = &v
+			}
+			var allowedValues *string = nil
+			if p.AllowedValues != nil {
+				v := *p.AllowedValues
+				allowedValues = &v
+			}
+			var dataType *string = nil
+			if p.DataType != nil {
+				v := *p.DataType
+				dataType = &v
+			}
+			var description *string = nil
+			if p.Description != nil {
+				v := *p.Description
+				description = &v
+			}
+			var isModifiable *bool = nil
+			if p.IsModifiable != nil {
+				v := *p.IsModifiable
+				isModifiable = &v
+			}
+			var minimumEngineVersion *string = nil
+			if p.MinimumEngineVersion != nil {
+				v := *p.MinimumEngineVersion
+				minimumEngineVersion = &v
+			}
+			var source *string = nil
+			if p.Source != nil {
+				v := *p.Source
+				source = &v
+			}
+			sp := svcapitypes.Parameter{
+				ParameterName:        name,
+				ParameterValue:       value,
+				Source:               source,
+				Description:          description,
+				IsModifiable:         isModifiable,
+				DataType:             dataType,
+				AllowedValues:        allowedValues,
+				MinimumEngineVersion: minimumEngineVersion,
```

Comment:
I'm curious to know why we copy the values instead of passing the pointers directly.

Reply:
Can something like this work? :
```go
sp := svcapitypes.Parameter{
	ParameterName:        p.ParameterName,
	ParameterValue:       p.ParameterValue,
	Source:               p.Source,
	Description:          p.Description,
	IsModifiable:         p.IsModifiable,
	DataType:             p.DataType,
	AllowedValues:        p.AllowedValues,
	MinimumEngineVersion: p.MinimumEngineVersion,
}
```

Reply:
It was not required. Updated code per the suggestion.